fashion-messaging-app-ui/src/components/Main/Main.jsx

```diff
@@ -20,20 +20,32 @@ function Main() {
       };
       fetchPosts();
     }, []);
-  
+
     const handleChange = (event) => {
-      setForm({
-        ...form,
-        [event.target.name]: event.target.value,
-      });
+      if (event.target.name === 'picture') {
+        setForm({
+          ...form,
+          [event.target.name]: event.target.files[0],
+        });
+      } else {
+        setForm({
+          ...form,
+          [event.target.name]: event.target.value,
+        });
+      }
     };
-  
+
     const handleSubmit = async (event) => {
       event.preventDefault();
+    
+      let formData = new FormData();
+      formData.append('title', form.title);
+      formData.append('content', form.content);
+      formData.append('picture', form.picture);
+    
       const response = await fetch('http://localhost:3000/posts', {
         method: 'POST',
-        headers: { 'Content-Type': 'application/json' },
-        body: JSON.stringify(form),
+        body: formData,
         credentials: 'include'
       });
       const newPost = await response.json();
```

Comment:
how are you handling the error that occurs if the user doesn't fill out everything in the form? could we handle it by showing an error message on the screen or something?

Reply:
> how are you handling the error that occurs if the user doesn't fill out everything in the form? could we handle it by showing an error message on the screen or something?

Okay, I will do that